src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/CallHandlerBase.cs

```diff
@@ -83,14 +85,25 @@ protected IList<BitVector> GetArguments(CilExecutionContext context, IMethodDesc
         {
             var stack = context.CurrentFrame.EvaluationStack;
             var result = new List<BitVector>(method.Signature!.GetTotalParameterCount());
+            var genericContext = GenericContext.FromMethod(method);
             
             // Pop sentinel arguments.
             for (int i = method.Signature.SentinelParameterTypes.Count - 1; i >= 0; i--)
-                result.Add(stack.Pop(method.Signature.SentinelParameterTypes[i]));
+            {
+                var sentinelParameter = method.Signature.SentinelParameterTypes[i];
+                result.Add(stack.Pop(sentinelParameter is GenericParameterSignature genericSentinelParameter
```

Comment:
Same here, the simple type check will not work for parameter types like `T[]` where `T` is the generic parameter.